{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ClinVar variant summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gn/1dwlg7ns6s3d_22hhl8kmrwm0000gn/T/ipykernel_5697/1986549347.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(\"/Users/ninaxiong/projects/orientdb/clinvar/variant_summary.txt\", sep=\"\\t\")\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(\"/Users/ninaxiong/projects/orientdb/clinvar/variant_summary.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# columns = [\"Type\", \"Name\", \"GeneSymbol\", \"ClinicalSignificance\", \"PhenotypeList\", \"Chromosome\", \"Start\", \"Stop\"]\n",
    "# df.loc[:5, columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Indel', 'Deletion', 'single nucleotide variant', 'Duplication',\n",
       "       'Microsatellite', 'Insertion', 'Variation', 'Translocation',\n",
       "       'protein only', 'Inversion', 'copy number gain', 'fusion',\n",
       "       'copy number loss', 'Complex', 'Tandem duplication'], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"Type\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39840</th>\n",
       "      <td>40616</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>GRCh38/hg38 10q11.22(chr10:46157935-47923579)x1</td>\n",
       "      <td>-1</td>\n",
       "      <td>covers 26 genes, none of which curated to show...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 18, 2013</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv575638,dbVar:nssv575725,dbVar:nssv57...</td>\n",
       "      <td>2</td>\n",
       "      <td>31951</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39841</th>\n",
       "      <td>40616</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>GRCh38/hg38 10q11.22(chr10:46157935-47923579)x1</td>\n",
       "      <td>-1</td>\n",
       "      <td>covers 26 genes, none of which curated to show...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 18, 2013</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv575638,dbVar:nssv575725,dbVar:nssv57...</td>\n",
       "      <td>2</td>\n",
       "      <td>31951</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39842</th>\n",
       "      <td>40617</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>GRCh38/hg38 2q37.3(chr2:241988449-242065208)x1</td>\n",
       "      <td>-1</td>\n",
       "      <td>LINC01237;LINC01880;LOC122889016;LOC285097</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 18, 2013</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv3395725,dbVar:nssv3395747,dbVar:nssv...</td>\n",
       "      <td>2</td>\n",
       "      <td>31952</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39843</th>\n",
       "      <td>40617</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>GRCh38/hg38 2q37.3(chr2:241988449-242065208)x1</td>\n",
       "      <td>-1</td>\n",
       "      <td>LINC01237;LINC01880;LOC122889016;LOC285097</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>Sep 18, 2013</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv3395725,dbVar:nssv3395747,dbVar:nssv...</td>\n",
       "      <td>2</td>\n",
       "      <td>31952</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39844</th>\n",
       "      <td>40621</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>GRCh38/hg38 16p11.2(chr16:29581462-30179247)x1</td>\n",
       "      <td>-1</td>\n",
       "      <td>subset of 37 genes: PRRT2</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 05, 2011</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv582635,dbVar:nssv582769,dbVar:nssv58...</td>\n",
       "      <td>2</td>\n",
       "      <td>31956</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #AlleleID              Type  \\\n",
       "39840      40616  copy number loss   \n",
       "39841      40616  copy number loss   \n",
       "39842      40617  copy number loss   \n",
       "39843      40617  copy number loss   \n",
       "39844      40621  copy number loss   \n",
       "\n",
       "                                                  Name  GeneID  \\\n",
       "39840  GRCh38/hg38 10q11.22(chr10:46157935-47923579)x1      -1   \n",
       "39841  GRCh38/hg38 10q11.22(chr10:46157935-47923579)x1      -1   \n",
       "39842   GRCh38/hg38 2q37.3(chr2:241988449-242065208)x1      -1   \n",
       "39843   GRCh38/hg38 2q37.3(chr2:241988449-242065208)x1      -1   \n",
       "39844   GRCh38/hg38 16p11.2(chr16:29581462-30179247)x1      -1   \n",
       "\n",
       "                                              GeneSymbol HGNC_ID  \\\n",
       "39840  covers 26 genes, none of which curated to show...       -   \n",
       "39841  covers 26 genes, none of which curated to show...       -   \n",
       "39842         LINC01237;LINC01880;LOC122889016;LOC285097       -   \n",
       "39843         LINC01237;LINC01880;LOC122889016;LOC285097       -   \n",
       "39844                          subset of 37 genes: PRRT2       -   \n",
       "\n",
       "      ClinicalSignificance  ClinSigSimple LastEvaluated  RS# (dbSNP)  ...  \\\n",
       "39840               Benign              0  Sep 18, 2013           -1  ...   \n",
       "39841               Benign              0  Sep 18, 2013           -1  ...   \n",
       "39842               Benign              0  Sep 18, 2013           -1  ...   \n",
       "39843               Benign              0  Sep 18, 2013           -1  ...   \n",
       "39844           Pathogenic              1  Aug 05, 2011           -1  ...   \n",
       "\n",
       "                         ReviewStatus NumberSubmitters Guidelines TestedInGTR  \\\n",
       "39840  no assertion criteria provided                1          -           N   \n",
       "39841  no assertion criteria provided                1          -           N   \n",
       "39842  no assertion criteria provided                4          -           N   \n",
       "39843  no assertion criteria provided                4          -           N   \n",
       "39844  no assertion criteria provided                2          -           N   \n",
       "\n",
       "                                                OtherIDs SubmitterCategories  \\\n",
       "39840  dbVar:nssv575638,dbVar:nssv575725,dbVar:nssv57...                   2   \n",
       "39841  dbVar:nssv575638,dbVar:nssv575725,dbVar:nssv57...                   2   \n",
       "39842  dbVar:nssv3395725,dbVar:nssv3395747,dbVar:nssv...                   2   \n",
       "39843  dbVar:nssv3395725,dbVar:nssv3395747,dbVar:nssv...                   2   \n",
       "39844  dbVar:nssv582635,dbVar:nssv582769,dbVar:nssv58...                   2   \n",
       "\n",
       "      VariationID PositionVCF ReferenceAlleleVCF  AlternateAlleleVCF  \n",
       "39840       31951          -1                 na                  na  \n",
       "39841       31951          -1                 na                  na  \n",
       "39842       31952          -1                 na                  na  \n",
       "39843       31952          -1                 na                  na  \n",
       "39844       31956          -1                 na                  na  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"Type\"] == \"copy number loss\"].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for x in df[\"ClinicalSignificance\"].unique():\n",
    "#     print(x)"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "243639d3113cbe381352a32fe5a4e1a317b20dd77d7d45bde22354ae8ddc7d11"
  },
  "kernelspec": {
   "display_name": "Python 3.9.7 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
